actions/restore.py

```diff
@@ -227,7 +246,14 @@ def dismantle_cluster():
     for name, data in etcdctl.member_list(endpoint).items():
         if name != my_name:
             log('Disconnecting {}'.format(name), hookenv.DEBUG)
-            etcdctl.unregister(data['unit_id'], endpoint)
+            for _ in range(5):
+                try:
+                    etcdctl.unregister(data['unit_id'], endpoint)
+                    break
+                except EtcdCtl.CommandFailed:
+                    # Back-off timer to let cluster settle
+                    log("Disconnecting {} failed, retrying...".format(name), "WARNING")
+                    time.sleep(_ + 1)
```

Comment:
Couple thoughts on this loop.  First, it shares a similar problem as the loop above -- it doesn't do anything special if we exhaust `range(5)`.  Shouldn't we raise in that case, or is it safe to continue rendering a conf even though `unregister` has failed? If it *is* safe, i think we should at least log in the last iteration that "disconnect failed, rendering anyway" or something to that effect.

Second, it's a little odd to have 2 loop styles so close together.  Either use a counter var (like the while loop above) or a range (like the for loop here) and be consistent with the sleeps (constant sleep or backoff, pick one).  This isn't a blocker, btw, just a note that consistent logic will be easier on future maintainers.